azurerm/internal/services/cdn/cdn_endpoint_resource.go

```diff
@@ -447,6 +445,10 @@ func resourceArmCdnEndpointRead(d *schema.ResourceData, meta interface{}) error
 		d.Set("probe_path", props.ProbePath)
 		d.Set("optimization_type", string(props.OptimizationType))
 
+		if is_compression_enabled := props.IsCompressionEnabled; is_compression_enabled != nil {
+			d.Set("is_compression_enabled", *is_compression_enabled)
+		}
+
```

Comment:
In order to get this to work as expected I suspect you are going to have to modify the `resourceArmCdnEndpointCreate` function as well. The reason for this is since the data type of `is_compression_enabled` is `bool` and in the original create function has this line of code:
```
compressionEnabled := d.Get("is_compression_enabled").(bool)
```
So even if the `d.Get` call does not find the `is_compression_enabled` attribute in the configuration file the value of `compressionEnabled` will be `false` because that is the default zero value for `bool`. Which means no matter what the `is_compression_enabled` attribute is, or is not, it will always be set to `false`. To fix this I think you will need to remove this line of code:
```
compressionEnabled := d.Get("is_compression_enabled").(bool)
``` 
and the this line of code from the `cdn.Endpoint` struct definition:
```
IsCompressionEnabled = &compressionEnabled
```
Then switch the logic in the `resourceArmCdnEndpointCreate` function to use the `d.GetOk` instead of just `d.Get` which will only set the value of `compressionEnabled` if the attribute actually exists in the configuration file, something along the lines of:

```
location := azure.NormalizeLocation(d.Get("location").(string))
httpAllowed := d.Get("is_http_allowed").(bool)
httpsAllowed := d.Get("is_https_allowed").(bool)
cachingBehaviour := d.Get("querystring_caching_behaviour").(string)
originHostHeader := d.Get("origin_host_header").(string)
originPath := d.Get("origin_path").(string)
probePath := d.Get("probe_path").(string)
optimizationType := d.Get("optimization_type").(string)
contentTypes := expandArmCdnEndpointContentTypesToCompress(d)
geoFilters := expandCdnEndpointGeoFilters(d)

t := d.Get("tags").(map[string]interface{})

endpoint := cdn.Endpoint{
	Location: &location,
	EndpointProperties: &cdn.EndpointProperties{
		ContentTypesToCompress:     &contentTypes,
		GeoFilters:                 geoFilters,
		IsHTTPAllowed:              &httpAllowed,
		IsHTTPSAllowed:             &httpsAllowed,
		QueryStringCachingBehavior: cdn.QueryStringCachingBehavior(cachingBehaviour),
		OriginHostHeader:           utils.String(originHostHeader),
	},
	Tags: tags.Expand(t),
}

if v, ok := d.GetOk("is_compression_enabled"); ok {
	endpoint.EndpointProperties.IsCompressionEnabled = &v.(bool)
}
```
Does that make sense? Also, can you please add a test case for this scenario where the `is_compression_enabled` attribute is not passed in the configuration file?